src/nn/prepo.rs

```diff
@@ -14,99 +19,109 @@ extern "C" {
         this: *mut PlayReport,
         arg1: *const u8,
     ) -> root::Result;
-}
-extern "C" {
+
     #[link_name = "\u{1}_ZN2nn5prepo10PlayReport9SetBufferEv"]
     pub fn PlayReport_SetBuffer(this: *mut PlayReport)
         -> root::Result;
-}
-extern "C" {
+
     #[link_name = "\u{1}_ZN2nn5prepo10PlayReport3AddEPKcl"]
     pub fn PlayReport_Add(
         this: *mut PlayReport,
-        arg1: *const u8,
-        arg2: i64,
+        key: *const u8,
+        value: i64,
     ) -> root::Result;
-}
-extern "C" {
+
     #[link_name = "\u{1}_ZN2nn5prepo10PlayReport3AddEPKcd"]
     pub fn PlayReport_Add1(
         this: *mut PlayReport,
-        arg1: *const u8,
-        arg2: f64,
+        key: *const u8,
+        value: f64,
     ) -> root::Result;
-}
-extern "C" {
+
     #[link_name = "\u{1}_ZN2nn5prepo10PlayReport3AddEPKcS3_"]
     pub fn PlayReport_Add2(
         this: *mut PlayReport,
-        arg1: *const u8,
-        arg2: *const u8,
+        key: *const u8,
+        value: *const u8,
     ) -> root::Result;
-}
-extern "C" {
+
+    #[link_name = "\u{1}_ZN2nn5prepo10PlayReport3AddEPKcRKNS0_10Any64BitIdE"]
+    pub fn PlayReport_Add3(
+            this: *mut PlayReport,
+            key: *const u8,
+            value: Any64BitId,
+    ) -> root::Result;
+
     #[link_name = "\u{1}_ZN2nn5prepo10PlayReport4SaveEv"]
     pub fn PlayReport_Save(this: *mut PlayReport) -> root::Result;
-}
-extern "C" {
+
     #[link_name = "\u{1}_ZN2nn5prepo10PlayReport4SaveERKNS_7account3UidE"]
     pub fn PlayReport_Save1(
         this: *mut PlayReport,
-        arg1: *const root::nn::account::Uid,
+        uid: *const root::nn::account::Uid,
     ) -> root::Result;
-}
-extern "C" {
+
     #[link_name = "\u{1}_ZN2nn5prepo10PlayReportC1Ev"]
     pub fn PlayReport_PlayReport(this: *mut PlayReport);
 }
 impl PlayReport {
     #[inline]
-    pub unsafe fn SetEventId(&mut self, arg1: *const u8) -> root::Result {
-        PlayReport_SetEventId(self, arg1)
+    pub fn SetEventId(&mut self, arg1: *const u8) -> root::Result {
+        unsafe { PlayReport_SetEventId(self, arg1) }
     }
     #[inline]
-    pub unsafe fn SetBuffer(&mut self) -> root::Result {
-        PlayReport_SetBuffer(self)
+    pub fn SetBuffer(&mut self) -> root::Result {
+        unsafe { PlayReport_SetBuffer(self) }
     }
     #[inline]
-    pub unsafe fn Add(
+    pub fn AddLong(
         &mut self,
-        arg1: *const u8,
-        arg2: i64,
+        key: *const u8,
```

Comment:
If you make Rust wrappers, ask for Rust arguments.
Key should be a String or &str or a generic type that converts to them (AsRef, Into, ...).
Make sure to append a \0 to what the user provide, as nnsdk expects a C string.